File: truss-compiler/src/main/java/truss/compiler/symbols/TypeSymbolBase.java
package truss.compiler.symbols;

public abstract class TypeSymbolBase extends ContainerSymbol {
    private final String name;
    private final String metadataName;

    protected TypeSymbolBase(String name, String metadataName, ContainerSymbol parent) {
        super(parent);

        this.name = name;
        this.metadataName = metadataName;
    }

    @Override
    public String getMetadataName() {
        return metadataName;
    }

    @Override
    public String getName() {
        return name;
    }

    public abstract TypeKind getTypeKind();
}

File: Truss.Compiler/Symbols/TypeSymbolBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public abstract class TypeSymbolBase : ContainerSymbol {
        private readonly string _name;
        private readonly string _metadataName;

        protected TypeSymbolBase(string name, string metadataName, ContainerSymbol parent)
            : base(parent) {
            _name = name;
            _metadataName = metadataName;
        }

        public override string Name {
            get { return _name; }
        }

        public override string MetadataName {
            get { return _metadataName; }
        }

        public abstract TypeKind TypeKind { get; }
    }
}
